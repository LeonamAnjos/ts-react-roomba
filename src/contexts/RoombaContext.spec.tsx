import { render, screen, fireEvent } from "@testing-library/react";
import { useContext } from "react";
import RoombaContextProvider, { RoombaContext } from "./RoombaContext";

type CustomRenderProps = {
  size: number;
};

const TestContext = () => {
  const {
    position: { row, column, size, direction },
    getEmoji,
    moveForward,
    rotateRight,
  } = useContext(RoombaContext);

  return (
    <>
      {`Row: '${row}' | Column: '${column}' | Size: '${size}' | Direction: '${direction}' | Emoji: '${getEmoji()}'`}
      <button data-testid={"move-forward-button"} onClick={moveForward} />
      <button data-testid={"rotate-right-button"} onClick={rotateRight} />
    </>
  );
};

describe("<RoombaContextProvier />", () => {
  const customRender = ({ size }: CustomRenderProps) => {
    return render(
      <RoombaContextProvider size={size}>
        <TestContext />
      </RoombaContextProvider>
    );
  };
  it("SHOULD render WHEN inicial context", () => {
    const { container } = customRender({ size: 3 });

    expect(container).toMatchSnapshot();
  });

  it("moveForward()", () => {
    const { container } = customRender({ size: 2 });
    const element = screen.getByTestId("move-forward-button");

    Array.from({ length: 8 }).forEach(() => {
      fireEvent.click(element);
      expect(container).toMatchSnapshot();
    });
  });

  it("rotateRight()", () => {
    const { container } = customRender({ size: 3 });
    const element = screen.getByTestId("rotate-right-button");

    Array.from({ length: 4 }).forEach(() => {
      fireEvent.click(element);
      expect(container).toMatchSnapshot();
    });
  });
});
