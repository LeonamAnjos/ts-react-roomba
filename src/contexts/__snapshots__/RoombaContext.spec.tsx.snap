// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoombaContextProvier /> SHOULD render WHEN inicial context 1`] = `
<div>
  Row: '0' | Column: '0' | Size: '3' | Direction: '1' | Emoji: 'ðŸ‘‰'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> moveForward() 1`] = `
<div>
  Row: '0' | Column: '1' | Size: '2' | Direction: '1' | Emoji: 'ðŸ‘‰'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> moveForward() 2`] = `
<div>
  Row: '0' | Column: '1' | Size: '2' | Direction: '2' | Emoji: 'ðŸ‘‡'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> moveForward() 3`] = `
<div>
  Row: '1' | Column: '1' | Size: '2' | Direction: '2' | Emoji: 'ðŸ‘‡'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> moveForward() 4`] = `
<div>
  Row: '1' | Column: '1' | Size: '2' | Direction: '3' | Emoji: 'ðŸ‘ˆ'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> moveForward() 5`] = `
<div>
  Row: '1' | Column: '0' | Size: '2' | Direction: '3' | Emoji: 'ðŸ‘ˆ'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> moveForward() 6`] = `
<div>
  Row: '1' | Column: '0' | Size: '2' | Direction: '4' | Emoji: 'ðŸ‘†'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> moveForward() 7`] = `
<div>
  Row: '0' | Column: '0' | Size: '2' | Direction: '4' | Emoji: 'ðŸ‘†'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> moveForward() 8`] = `
<div>
  Row: '0' | Column: '0' | Size: '2' | Direction: '1' | Emoji: 'ðŸ‘‰'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> rotateRight() 1`] = `
<div>
  Row: '0' | Column: '0' | Size: '3' | Direction: '2' | Emoji: 'ðŸ‘‡'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> rotateRight() 2`] = `
<div>
  Row: '0' | Column: '0' | Size: '3' | Direction: '3' | Emoji: 'ðŸ‘ˆ'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> rotateRight() 3`] = `
<div>
  Row: '0' | Column: '0' | Size: '3' | Direction: '4' | Emoji: 'ðŸ‘†'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;

exports[`<RoombaContextProvier /> rotateRight() 4`] = `
<div>
  Row: '0' | Column: '0' | Size: '3' | Direction: '1' | Emoji: 'ðŸ‘‰'
  <button
    data-testid="move-forward-button"
  />
  <button
    data-testid="rotate-right-button"
  />
</div>
`;
